src/amperity/gocd/secret/vault/VaultSecretsPlugin.java

```diff
@@ -22,7 +21,7 @@
 
     private GoApplicationAccessor accessor;  // Set of JSON API's exposing GoCD info specifically curated for plugins.
     private IFn handler;  // Exposes plugin API
-    private Atom state;
+    private IFn client;  // The Vault Client, TODO: IMPLEMENT, currently in inconsistent state as an atom
```

Comment:
The type here should probably be the interface generated by the vault client protocol, but since you're just going to be passing this to Clojure it doesn't really matter.

Reply:
@greglook Do you have some resources I could check out to learn more about this? Most of the resources I've seen on Java interop only go into the very basics

Reply:
https://clojure.org/reference/protocols - `defprotocol` produces a regular Java interface as part of its output, so you could declare that this variable has the interface as the type. However, unless you plan to call any methods on this from Java, it'd be simpler to just call `client` an `Object`.